rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -705,20 +843,47 @@ def _create_model_data(
         )
 
         model_data = RasaModelData(
-            label_key=self.label_key, label_sub_key=self.label_sub_key
-        )
+            label_key=self.label_key, label_sub_key=self.label_sub_key,
+        )  # TODO: rest of the code uses/used: LABEL_KEY/LABEL_SUB_LEY or LABEL/ID
         model_data.add_data(attribute_data)
+
+        # the following will only add a sequence length if SEQUENCE sub_key exists
         model_data.add_lengths(TEXT, SEQUENCE_LENGTH, TEXT, SEQUENCE)
+
         # Current implementation doesn't yet account for updating sparse
         # feature sizes of label attributes. That's why we remove them.
         sparse_feature_sizes = self._remove_label_sparse_feature_sizes(
-            sparse_feature_sizes=sparse_feature_sizes, label_attribute=label_attribute
+            sparse_feature_sizes=sparse_feature_sizes,
+            label_attribute=self.label_attribute,
         )
         model_data.add_sparse_feature_sizes(sparse_feature_sizes)
 
-        self._add_label_features(
-            model_data, training_data, label_attribute, label_id_dict, training
-        )
+        label_ids = []
+        if training and self.component_config[INTENT_CLASSIFICATION]:
+            label_ids = self._add_label_id_features(model_data, messages, label_id_dict)
+
+            # if no label features have been found before (see above RasaModel
+            # creation), then we load the default label features that should've been
+            # computed by an self._create_label_data() call (cf. check at the beginning
+            # of this function).
+            if model_data.does_feature_not_exist(
+                self.label_attribute, SENTENCE
+            ) and model_data.does_feature_not_exist(self.label_attribute, SEQUENCE):
+                # no label features are present, get default features from _label_data
+                model_data.add_features(
+                    LABEL_KEY,
+                    SENTENCE,
+                    self._use_default_label_features(np.array(label_ids)),
+                )
+
+            # As label_attribute can have different values, e.g. INTENT or RESPONSE,
+            # copy over the features to the LABEL key to make
+            # it easier to access the label features inside the model itself.
+            # Note that update_key doesn't update any keys that aren't there.
+            for subkey in [SENTENCE, SEQUENCE_LENGTH, MASK, SEQUENCE]:
+                model_data.update_key(self.label_attribute, subkey, LABEL_KEY, subkey)
```

Comment:
In earlier comments I thought I can and should just let label_key() output the correct label and then drop the copying. But RasaModel does some mapping as well:  https://github.com/RasaHQ/rasa/blob/main/rasa/utils/tensorflow/models.py#L693 . Which means all tf models would be affected if I start changing that. So now I'm thinking it's easier to just remove "label_key()" to avoid further confusion :D.